pytensor/link/pytorch/dispatch/subtensor.py

```diff
@@ -0,0 +1,111 @@
+from pytensor.link.pytorch.dispatch.basic import pytorch_funcify
+from pytensor.tensor.subtensor import (
+    AdvancedIncSubtensor,
+    AdvancedIncSubtensor1,
+    AdvancedSubtensor,
+    AdvancedSubtensor1,
+    IncSubtensor,
+    Subtensor,
+    indices_from_subtensor,
+)
+from pytensor.tensor.type_other import MakeSlice
+
+
+def check_negative_steps(index):
+    for i in index:
+        if isinstance(i, slice):
+            if i.step and isinstance(i.step, int) and i.step < 0:
+                raise NotImplementedError(
+                    "Negative step sizes are not supported in Pytorch"
+                )
+            elif i.step and not isinstance(i.step, int):
+                raise NotImplementedError(
+                    "Negative step sizes are not supported in Pytorch"
+                )
+
+
+@pytorch_funcify.register(Subtensor)
+def pytorch_funcify_Subtensor(op, node, **kwargs):
+    idx_list = getattr(op, "idx_list", None)
+
+    def subtensor(x, *ilists):
+        indices = indices_from_subtensor(ilists, idx_list)
+        check_negative_steps(indices)
+
+        return x[indices]
+
+    return subtensor
+
+
+@pytorch_funcify.register(AdvancedSubtensor1)
+@pytorch_funcify.register(AdvancedSubtensor)
+def pytorch_funcify_AdvSubtensor(op, node, **kwargs):
+    def advsubtensor(x, *indices):
+        check_negative_steps(indices)
+        return x[indices]
+
+    return advsubtensor
+
+
+@pytorch_funcify.register(MakeSlice)
+def pytorch_funcify_makeslice(op, **kwargs):
+    def makeslice(*x):
+        return slice(x)
+
+    return makeslice
+
+
+@pytorch_funcify.register(IncSubtensor)
+def pytorch_funcify_IncSubtensor(op, node, **kwargs):
+    idx_list = getattr(op, "idx_list", None)
+
+    if getattr(op, "set_instead_of_inc", False):
+
+        def torch_fn(x, indices, y):
+            check_negative_steps(indices)
+            x[indices] = y
+            return x
+
+    else:
+
+        def torch_fn(x, indices, y):
+            check_negative_steps(indices)
+            x1 = x.clone()
+            x1[indices] += y
+            return x1
+
+    def incsubtensor(x, y, *ilist, torch_fn=torch_fn, idx_list=idx_list):
+        indices = indices_from_subtensor(ilist, idx_list)
+        if len(indices) == 1:
+            indices = indices[0]
```

Comment:
Why is this needed? Can't we just call  `x[tuple(indices)]` regardless of whether its one or more?